subsys/net/lib/nrf_cloud/src/nrf_cloud_logs.c

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <net/nrf_cloud.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_ctrl.h>
+#include <zephyr/logging/log_output.h>
+#include <zephyr/logging/log_backend.h>
+#include <zephyr/sys/ring_buffer.h>
+#include <zephyr/sys/base64.h>
+#include <date_time.h>
+#include "nrf_cloud_fsm.h"
+#include "nrf_cloud_mem.h"
+#include "nrf_cloud_codec_internal.h"
+#include "nrf_cloud_transport.h"
+#include "nrf_cloud_logs_internal.h"
+#include <net/nrf_cloud_rest.h>
+#include <net/nrf_cloud_logs.h>
+
+#define REST_WAIT_MS 1000
+
+LOG_MODULE_REGISTER(nrf_cloud_logs, CONFIG_NRF_CLOUD_LOGGER_LOG_LEVEL);
+
+static bool enabled;
+static int nrf_cloud_log_level = CONFIG_NRF_CLOUD_LOG_BACKEND_LEVEL;
+
+#if defined(CONFIG_NRF_CLOUD_MQTT) || defined(CONFIG_NRF_CLOUD_REST)
+static int64_t starting_timestamp;
+static atomic_t log_sequence;
+#endif
+
+static K_SEM_DEFINE(ncl_active, 1, 1);
+
+void nrf_cloud_log_init(void)
+{
+	static bool initialized;
+	int err = 0;
+
+	if (initialized) {
+		return;
+	}
+
+	(void)nrf_cloud_codec_init(NULL);
+
+	if (IS_ENABLED(CONFIG_DATE_TIME) && !starting_timestamp) {
```

Comment:
Is the `starting_timestamp` check needed? It should be zero when not yet initialized and non-zero if initialization has succeeded.

Reply:
good point